### ISSUE TITLE
Slacker API

### ISSUE THREAD

**Issue by jaspermayone** [hackclub/slacker#172]
Welcome team is building a more custom solution for welcomes, but would still love to use slacker for backend tracking (or perhaps just elastic?) 

Are their Slacker API routes we can tie into? (I think @grymmy said this was in the backlog?)

**Comment by grymmy** [comment]
Slacker doesn't just provide backend tracking - critically, it provides a work queue for action items.  Whatever tool you build, I highly recommend integration fully into both the "pull from work queue" pattern that Slacker provides on the backend, and also the backend tracking that it provides for free.  TBH, I feel that your plan to build a new separate tool for the welcomers use case (instead of working on modifying slacker via feature requests to conform more to what you want, via config) is a bad idea at its core, leading to a more fragmented ops environment - I've made my stance on this pretty clear in the past couple weeks.

That being said, providing an API **is** a reasonable request.  Please provide a complete list of backend slacker APIs that you need to accomplish what you want.

**Comment by jaspermayone** [comment]
Hey, as a starters wanted to reply to @grymmy's comments on his stance with the Professor Bloom project.

Here is the message I sent to the welcome team earlier today. 
![CleanShot 2024-04-14 at 16 43 31@2x](https://github.com/hackclub/slacker/assets/65788728/57bf4456-97d9-45b4-b547-33d930a9b069)

TLDR: We will still be using Slacker, the front end is what is changing. (Hence the request for an API)

**Comment by jaspermayone** [comment]
My thought behind creating something separate is mainly to keep Slacker general purpose, while allowing for a welcomer specific tool.

**Comment by jaspermayone** [comment]
In terms of an API,

Routes for modifying assignee, changing tix status, and just about any other slacker workflow action would be great.

**Comment by faisalsayed10** [comment]
@jaspermayone as per our conversation, we decided that you will be giving me a clear and concise list of all the apis that you need, assigning back to you :))

**Comment by jaspermayone** [comment]
Here is a full list of endpoints needed.

- Create action item (returns actionid for future reference)
- Assign action item (takes in actionid)
- mark action as irrelevant (takes in actionid)
- resolve action item (takes in actionid)
- create / edit / delete action item notes
- create followup (and resolve although i would assume this can be bundled with the other resolve endpoint?) (takes in actionid)
-

**Comment by jaspermayone** [comment]
@faisalsayed10 do you think I am missing anything key to the flow of slacker here?

**Comment by faisalsayed10** [comment]
apis have been shipped
<img width="824" alt="image" src="https://github.com/hackclub/slacker/assets/36627266/2b1822e9-5892-40eb-88ff-4775882ab2e8">

**Comment by jaspermayone** [comment]
LGTM, ty!
